pkg/sdk/stages_dto_builders_gen.go

```diff
@@ -0,0 +1,825 @@
+) *InternalStageEncryptionRequest {
+	s := InternalStageEncryptionRequest{}
+	s.Type = Type
+	return &s
+}
+
+func NewInternalDirectoryTableOptionsRequest() *InternalDirectoryTableOptionsRequest {
+	return &InternalDirectoryTableOptionsRequest{}
+}
+
+func (s *InternalDirectoryTableOptionsRequest) WithEnable(Enable *bool) *InternalDirectoryTableOptionsRequest {
+	s.Enable = Enable
+	return s
+}
+
+func (s *InternalDirectoryTableOptionsRequest) WithRefreshOnCreate(RefreshOnCreate *bool) *InternalDirectoryTableOptionsRequest {
+	s.RefreshOnCreate = RefreshOnCreate
+	return s
+}
+
+func NewStageFileFormatRequest() *StageFileFormatRequest {
+	return &StageFileFormatRequest{}
+}
+
+func (s *StageFileFormatRequest) WithFormatName(FormatName *string) *StageFileFormatRequest {
+	s.FormatName = FormatName
+	return s
+}
+
+func (s *StageFileFormatRequest) WithType(Type *FileFormatType) *StageFileFormatRequest {
+	s.Type = Type
+	return s
+}
+
+func (s *StageFileFormatRequest) WithOptions(Options *FileFormatTypeOptions) *StageFileFormatRequest {
+	s.Options = Options
+	return s
+}
+
+func NewStageCopyOptionsRequest() *StageCopyOptionsRequest {
+	return &StageCopyOptionsRequest{}
+}
+
+func (s *StageCopyOptionsRequest) WithOnError(OnError *StageCopyOnErrorOptionsRequest) *StageCopyOptionsRequest {
+	s.OnError = OnError
+	return s
+}
+
+func (s *StageCopyOptionsRequest) WithSizeLimit(SizeLimit *int) *StageCopyOptionsRequest {
+	s.SizeLimit = SizeLimit
+	return s
+}
+
+func (s *StageCopyOptionsRequest) WithPurge(Purge *bool) *StageCopyOptionsRequest {
+	s.Purge = Purge
+	return s
+}
+
+func (s *StageCopyOptionsRequest) WithReturnFailedOnly(ReturnFailedOnly *bool) *StageCopyOptionsRequest {
+	s.ReturnFailedOnly = ReturnFailedOnly
+	return s
+}
+
+func (s *StageCopyOptionsRequest) WithMatchByColumnName(MatchByColumnName *StageCopyColumnMapOption) *StageCopyOptionsRequest {
+	s.MatchByColumnName = MatchByColumnName
+	return s
+}
+
+func (s *StageCopyOptionsRequest) WithEnforceLength(EnforceLength *bool) *StageCopyOptionsRequest {
+	s.EnforceLength = EnforceLength
+	return s
+}
+
+func (s *StageCopyOptionsRequest) WithTruncatecolumns(Truncatecolumns *bool) *StageCopyOptionsRequest {
+	s.Truncatecolumns = Truncatecolumns
+	return s
+}
+
+func (s *StageCopyOptionsRequest) WithForce(Force *bool) *StageCopyOptionsRequest {
+	s.Force = Force
+	return s
+}
+
+func NewStageCopyOnErrorOptionsRequest() *StageCopyOnErrorOptionsRequest {
+	return &StageCopyOnErrorOptionsRequest{}
+}
+
+func (s *StageCopyOnErrorOptionsRequest) WithContinue(Continue *bool) *StageCopyOnErrorOptionsRequest {
+	s.Continue = Continue
+	return s
+}
+
+func (s *StageCopyOnErrorOptionsRequest) WithSkipFile(SkipFile *bool) *StageCopyOnErrorOptionsRequest {
+	s.SkipFile = SkipFile
+	return s
+}
+
+func (s *StageCopyOnErrorOptionsRequest) WithAbortStatement(AbortStatement *bool) *StageCopyOnErrorOptionsRequest {
+	s.AbortStatement = AbortStatement
+	return s
+}
+
+func NewCreateOnS3StageRequest(
```

Comment:
if you have `CreateInternal` then wouldnt it be more appropriate to call this `CreateExternalForS3`?

Reply:
Same as for enum types, that could be our next generator improvement. Added to readme. (previous discussion on this - https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2113#discussion_r1356777447)